apps/condo/domains/news/schema/ExportNewsRecipientsService.js

```diff
@@ -0,0 +1,205 @@
+                                id: organizationId,
+                            },
+                            deletedAt: null,
+                        },
+                    })
+
+                    for (let property of propertiesByOrganization) {
+                        const units = getUnitsFromProperty({ map: property.map })
+
+                        const recipientsData = units.reduce((acc, unit) => {
+                            if (residentsByOrganization.find((resident) => unit.unitName === resident.unitName)) {
+                                acc.push({ address: property.address, unitName: unit.unitName, hasResident: true })
+                                return [...acc]
+                            }
+                            acc.push({ address: property.address, unitName: unit.unitName, hasResident: false })
+                            return [...acc]
+                        }, [])
+                        recipientsByOrganization.push(...recipientsData)
+                    }
+                }
+
+                const recipientsByProperty = []
+                const recipientsByUnitType = []
+                const recipientsByUnitName = []
+                if (!isEmpty(newsItemScopes)) {
+                    for (let newsItemScope of newsItemScopes) {
+                        if (get(newsItemScope, 'property.id')) {
+                            const property = await Property.getOne(context, {
+                                id: newsItemScope.property.id,
+                                deletedAt: null,
+                            })
+
+                            const units = getUnitsFromProperty({ map: property.map })
+
+                            if (!newsItemScope.unitName && !newsItemScope.unitType) {
+                                const recipientsData = units.reduce((acc, unit) => {
+                                    if (residentsByProperties.find((resident) => unit.unitName === resident.unitName)) {
+                                        acc.push({ address: property.address, unitName: unit.unitName, hasResident: true })
+                                        return [...acc]
+                                    }
+                                    acc.push({ address: property.address, unitName: unit.unitName, hasResident: false })
+                                    return [...acc]
+                                }, [])
+                                recipientsByProperty.push(...recipientsData)
+                            }
+
+                            if (!newsItemScope.unitName && newsItemScope.unitType) {
+                                const unitsFilteredByType = units.reduce((acc, unit) => {
+                                    if (unit.unitType === newsItemScope.unitType) {
+                                        acc.push(unit)
+                                    }
+                                    return [...acc]
+                                }, [])
```

Comment:
It is not a reduction of one data array to some another array, it's a filtering.


```suggestion
                                const unitsFilteredByType = filter(units, { unitType: newsItemScope.unitType })
```

Reply:
Does it affect performance that much? I see no reason for CR. Changed in last commit